{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Explore differentially expressed genes**  \n",
    "This part will asssume that you have a list of differentially expressed genes already generated.  \n",
    "Lets start with reading in some data to our session"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "suppressMessages(library(DESeq2)) # load the DESeq2 package"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "counts <- read.delim(\"data/counts.txt\", sep=\"\\t\", check.names=FALSE, as.is=TRUE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "sample.info <- read.delim(\"data/sample_info.txt\", sep=\"\\t\", check.names=FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "model <- read.delim(\"data//diffexpr-results.csv\", sep=\"\\t\", check.names=FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<thead><tr><th></th><th scope=col>gene_name</th><th scope=col>gene_biotype</th><th scope=col>baseMean</th><th scope=col>log2FoldChange</th><th scope=col>lfcSE</th><th scope=col>stat</th><th scope=col>pvalue</th><th scope=col>padj</th></tr></thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>ENSG00000091972</th><td>CD200         </td><td>protein_coding</td><td> 308.88085    </td><td> 3.655514     </td><td>0.2549227     </td><td> 14.24852     </td><td>4.578529e-46  </td><td>8.232652e-42  </td></tr>\n",
       "\t<tr><th scope=row>ENSG00000187601</th><td>MAGEH1        </td><td>protein_coding</td><td> 502.70770    </td><td> 3.161066     </td><td>0.2713160     </td><td> 11.62730     </td><td>2.994322e-31  </td><td>2.692045e-27  </td></tr>\n",
       "\t<tr><th scope=row>ENSG00000235304</th><td>LINC01281     </td><td>lincRNA       </td><td>  82.04318    </td><td> 4.885393     </td><td>0.3725763     </td><td> 11.40383     </td><td>4.001298e-30  </td><td>2.398245e-26  </td></tr>\n",
       "\t<tr><th scope=row>ENSG00000125538</th><td>IL1B          </td><td>protein_coding</td><td>1089.90272    </td><td>-1.868674     </td><td>0.1713032     </td><td>-10.90638     </td><td>1.074530e-27  </td><td>4.830280e-24  </td></tr>\n",
       "\t<tr><th scope=row>ENSG00000003147</th><td>ICA1          </td><td>protein_coding</td><td> 449.93853    </td><td> 3.382331     </td><td>0.3211397     </td><td> 10.50644     </td><td>8.068215e-26  </td><td>2.901491e-22  </td></tr>\n",
       "\t<tr><th scope=row>ENSG00000149257</th><td>SERPINH1      </td><td>protein_coding</td><td>1233.86635    </td><td> 3.537105     </td><td>0.3401214     </td><td> 10.40155     </td><td>2.439450e-25  </td><td>7.310624e-22  </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "\\begin{tabular}{r|llllllll}\n",
       "  & gene\\_name & gene\\_biotype & baseMean & log2FoldChange & lfcSE & stat & pvalue & padj\\\\\n",
       "\\hline\n",
       "\tENSG00000091972 & CD200            & protein\\_coding &  308.88085       &  3.655514        & 0.2549227        &  14.24852        & 4.578529e-46     & 8.232652e-42    \\\\\n",
       "\tENSG00000187601 & MAGEH1           & protein\\_coding &  502.70770       &  3.161066        & 0.2713160        &  11.62730        & 2.994322e-31     & 2.692045e-27    \\\\\n",
       "\tENSG00000235304 & LINC01281      & lincRNA        &   82.04318     &  4.885393      & 0.3725763      &  11.40383      & 4.001298e-30   & 2.398245e-26  \\\\\n",
       "\tENSG00000125538 & IL1B             & protein\\_coding & 1089.90272       & -1.868674        & 0.1713032        & -10.90638        & 1.074530e-27     & 4.830280e-24    \\\\\n",
       "\tENSG00000003147 & ICA1             & protein\\_coding &  449.93853       &  3.382331        & 0.3211397        &  10.50644        & 8.068215e-26     & 2.901491e-22    \\\\\n",
       "\tENSG00000149257 & SERPINH1         & protein\\_coding & 1233.86635       &  3.537105        & 0.3401214        &  10.40155        & 2.439450e-25     & 7.310624e-22    \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "| <!--/--> | gene_name | gene_biotype | baseMean | log2FoldChange | lfcSE | stat | pvalue | padj |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| ENSG00000091972 | CD200          | protein_coding |  308.88085     |  3.655514      | 0.2549227      |  14.24852      | 4.578529e-46   | 8.232652e-42   |\n",
       "| ENSG00000187601 | MAGEH1         | protein_coding |  502.70770     |  3.161066      | 0.2713160      |  11.62730      | 2.994322e-31   | 2.692045e-27   |\n",
       "| ENSG00000235304 | LINC01281      | lincRNA        |   82.04318     |  4.885393      | 0.3725763      |  11.40383      | 4.001298e-30   | 2.398245e-26   |\n",
       "| ENSG00000125538 | IL1B           | protein_coding | 1089.90272     | -1.868674      | 0.1713032      | -10.90638      | 1.074530e-27   | 4.830280e-24   |\n",
       "| ENSG00000003147 | ICA1           | protein_coding |  449.93853     |  3.382331      | 0.3211397      |  10.50644      | 8.068215e-26   | 2.901491e-22   |\n",
       "| ENSG00000149257 | SERPINH1       | protein_coding | 1233.86635     |  3.537105      | 0.3401214      |  10.40155      | 2.439450e-25   | 7.310624e-22   |\n",
       "\n"
      ],
      "text/plain": [
       "                gene_name gene_biotype   baseMean   log2FoldChange lfcSE    \n",
       "ENSG00000091972 CD200     protein_coding  308.88085  3.655514      0.2549227\n",
       "ENSG00000187601 MAGEH1    protein_coding  502.70770  3.161066      0.2713160\n",
       "ENSG00000235304 LINC01281 lincRNA          82.04318  4.885393      0.3725763\n",
       "ENSG00000125538 IL1B      protein_coding 1089.90272 -1.868674      0.1713032\n",
       "ENSG00000003147 ICA1      protein_coding  449.93853  3.382331      0.3211397\n",
       "ENSG00000149257 SERPINH1  protein_coding 1233.86635  3.537105      0.3401214\n",
       "                stat      pvalue       padj        \n",
       "ENSG00000091972  14.24852 4.578529e-46 8.232652e-42\n",
       "ENSG00000187601  11.62730 2.994322e-31 2.692045e-27\n",
       "ENSG00000235304  11.40383 4.001298e-30 2.398245e-26\n",
       "ENSG00000125538 -10.90638 1.074530e-27 4.830280e-24\n",
       "ENSG00000003147  10.50644 8.068215e-26 2.901491e-22\n",
       "ENSG00000149257  10.40155 2.439450e-25 7.310624e-22"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(model)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Subset data to significant genes and variance transform data (rlog)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "<table width=\"100%\" summary=\"page for rlog {DESeq2}\"><tr><td>rlog {DESeq2}</td><td style=\"text-align: right;\">R Documentation</td></tr></table>\n",
       "\n",
       "<h2>Apply a 'regularized log' transformation</h2>\n",
       "\n",
       "<h3>Description</h3>\n",
       "\n",
       "<p>This function transforms the count data to the log2 scale in a way \n",
       "which minimizes differences between samples for rows with small counts,\n",
       "and which normalizes with respect to library size.\n",
       "The rlog transformation produces a similar variance stabilizing effect as\n",
       "<code>varianceStabilizingTransformation</code>,\n",
       "though <code>rlog</code> is more robust in the\n",
       "case when the size factors vary widely.\n",
       "The transformation is useful when checking for outliers\n",
       "or as input for machine learning techniques\n",
       "such as clustering or linear discriminant analysis.\n",
       "<code>rlog</code> takes as input a <code>DESeqDataSet</code> and returns a\n",
       "<code>RangedSummarizedExperiment</code> object.\n",
       "</p>\n",
       "\n",
       "\n",
       "<h3>Usage</h3>\n",
       "\n",
       "<pre>\n",
       "rlog(object, blind = TRUE, intercept, betaPriorVar,\n",
       "  fitType = \"parametric\")\n",
       "\n",
       "rlogTransformation(object, blind = TRUE, intercept, betaPriorVar,\n",
       "  fitType = \"parametric\")\n",
       "</pre>\n",
       "\n",
       "\n",
       "<h3>Arguments</h3>\n",
       "\n",
       "<table summary=\"R argblock\">\n",
       "<tr valign=\"top\"><td><code>object</code></td>\n",
       "<td>\n",
       "<p>a DESeqDataSet, or matrix of counts</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>blind</code></td>\n",
       "<td>\n",
       "<p>logical, whether to blind the transformation to the experimental\n",
       "design. blind=TRUE should be used for comparing samples in an manner unbiased by\n",
       "prior information on samples, for example to perform sample QA (quality assurance).\n",
       "blind=FALSE should be used for transforming data for downstream analysis,\n",
       "where the full use of the design information should be made.\n",
       "blind=FALSE will skip re-estimation of the dispersion trend, if this has already been calculated.\n",
       "If many of genes have large differences in counts due to\n",
       "the experimental design, it is important to set blind=FALSE for downstream\n",
       "analysis.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>intercept</code></td>\n",
       "<td>\n",
       "<p>by default, this is not provided and calculated automatically.\n",
       "if provided, this should be a vector as long as the number of rows of object,\n",
       "which is log2 of the mean normalized counts from a previous dataset.\n",
       "this will enforce the intercept for the GLM, allowing for a &quot;frozen&quot; rlog\n",
       "transformation based on a previous dataset.\n",
       "You will also need to provide <code>mcols(object)$dispFit</code>.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>betaPriorVar</code></td>\n",
       "<td>\n",
       "<p>a single value, the variance of the prior on the sample\n",
       "betas, which if missing is estimated from the data</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>fitType</code></td>\n",
       "<td>\n",
       "<p>in case dispersions have not yet been estimated for <code>object</code>,\n",
       "this parameter is passed on to <code>estimateDispersions</code> (options described there).</p>\n",
       "</td></tr>\n",
       "</table>\n",
       "\n",
       "\n",
       "<h3>Details</h3>\n",
       "\n",
       "<p>Note that neither rlog transformation nor the VST are used by the\n",
       "differential expression estimation in <code>DESeq</code>, which always\n",
       "occurs on the raw count data, through generalized linear modeling which\n",
       "incorporates knowledge of the variance-mean dependence. The rlog transformation\n",
       "and VST are offered as separate functionality which can be used for visualization,\n",
       "clustering or other machine learning tasks. See the transformation section of the\n",
       "vignette for more details, including a statement on timing. If <code>rlog</code>\n",
       "is run on data with number of samples in [30-49] it will print a message\n",
       "that it may take a few minutes, if the number of samples is 50 or larger, it\n",
       "will print a message that it may take a &quot;long time&quot;, and in both cases, it\n",
       "will mention that the <code>vst</code> is a much faster transformation.\n",
       "</p>\n",
       "<p>The transformation does not require that one has already estimated size factors\n",
       "and dispersions.\n",
       "</p>\n",
       "<p>The regularization is on the log fold changes of the count for each sample\n",
       "over an intercept, for each gene. As nearby count values for low counts genes\n",
       "are almost as likely as the observed count, the rlog shrinkage is greater for low counts.\n",
       "For high counts, the rlog shrinkage has a much weaker effect.\n",
       "The fitted dispersions are used rather than the MAP dispersions\n",
       "(so similar to the <code>varianceStabilizingTransformation</code>).\n",
       "</p>\n",
       "<p>The prior variance for the shrinkag of log fold changes is calculated as follows: \n",
       "a matrix is constructed of the logarithm of the counts plus a pseudocount of 0.5,\n",
       "the log of the row means is then subtracted, leaving an estimate of\n",
       "the log fold changes per sample over the fitted value using only an intercept.\n",
       "The prior variance is then calculated by matching the upper quantiles of the observed \n",
       "log fold change estimates with an upper quantile of the normal distribution.\n",
       "A GLM fit is then calculated using this prior. It is also possible to supply the variance of the prior.\n",
       "See the vignette for an example of the use and a comparison with <code>varianceStabilizingTransformation</code>.\n",
       "</p>\n",
       "<p>The transformed values, rlog(K), are equal to\n",
       "<i>rlog(K_ij) = log2(q_ij) = beta_i0 + beta_ij</i>,\n",
       "with formula terms defined in <code>DESeq</code>.\n",
       "</p>\n",
       "<p>The parameters of the rlog transformation from a previous dataset\n",
       "can be frozen and reapplied to new samples. See the 'Data quality assessment'\n",
       "section of the vignette for strategies to see if new samples are\n",
       "sufficiently similar to previous datasets. \n",
       "The frozen rlog is accomplished by saving the dispersion function,\n",
       "beta prior variance and the intercept from a previous dataset,\n",
       "and running <code>rlog</code> with 'blind' set to FALSE\n",
       "(see example below).\n",
       "</p>\n",
       "\n",
       "\n",
       "<h3>Value</h3>\n",
       "\n",
       "<p>a <code>DESeqTransform</code> if a <code>DESeqDataSet</code> was provided,\n",
       "or a matrix if a count matrix was provided as input.\n",
       "Note that for <code>DESeqTransform</code> output, the matrix of\n",
       "transformed values is stored in <code>assay(rld)</code>.\n",
       "To avoid returning matrices with NA values, in the case of a row\n",
       "of all zeros, the rlog transformation returns zeros\n",
       "(essentially adding a pseudocount of 1 only to these rows).\n",
       "</p>\n",
       "\n",
       "\n",
       "<h3>References</h3>\n",
       "\n",
       "<p>Reference for regularized logarithm (rlog):\n",
       "</p>\n",
       "<p>Michael I Love, Wolfgang Huber, Simon Anders: Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2. Genome Biology 2014, 15:550. <a href=\"http://dx.doi.org/10.1186/s13059-014-0550-8\">http://dx.doi.org/10.1186/s13059-014-0550-8</a>\n",
       "</p>\n",
       "\n",
       "\n",
       "<h3>See Also</h3>\n",
       "\n",
       "<p><code>plotPCA</code>, <code>varianceStabilizingTransformation</code>, <code>normTransform</code>\n",
       "</p>\n",
       "\n",
       "\n",
       "<h3>Examples</h3>\n",
       "\n",
       "<pre>\n",
       "\n",
       "dds &lt;- makeExampleDESeqDataSet(m=6,betaSD=1)\n",
       "rld &lt;- rlog(dds)\n",
       "dists &lt;- dist(t(assay(rld)))\n",
       "# plot(hclust(dists))\n",
       "\n",
       "</pre>\n",
       "\n",
       "<hr /><div style=\"text-align: center;\">[Package <em>DESeq2</em> version 1.22.1 ]</div>"
      ],
      "text/latex": [
       "\\inputencoding{utf8}\n",
       "\\HeaderA{rlog}{Apply a 'regularized log' transformation}{rlog}\n",
       "\\aliasA{rlogTransformation}{rlog}{rlogTransformation}\n",
       "%\n",
       "\\begin{Description}\\relax\n",
       "This function transforms the count data to the log2 scale in a way \n",
       "which minimizes differences between samples for rows with small counts,\n",
       "and which normalizes with respect to library size.\n",
       "The rlog transformation produces a similar variance stabilizing effect as\n",
       "\\code{\\LinkA{varianceStabilizingTransformation}{varianceStabilizingTransformation}},\n",
       "though \\code{rlog} is more robust in the\n",
       "case when the size factors vary widely.\n",
       "The transformation is useful when checking for outliers\n",
       "or as input for machine learning techniques\n",
       "such as clustering or linear discriminant analysis.\n",
       "\\code{rlog} takes as input a \\code{\\LinkA{DESeqDataSet}{DESeqDataSet}} and returns a\n",
       "\\code{\\LinkA{RangedSummarizedExperiment}{RangedSummarizedExperiment}} object.\n",
       "\\end{Description}\n",
       "%\n",
       "\\begin{Usage}\n",
       "\\begin{verbatim}\n",
       "rlog(object, blind = TRUE, intercept, betaPriorVar,\n",
       "  fitType = \"parametric\")\n",
       "\n",
       "rlogTransformation(object, blind = TRUE, intercept, betaPriorVar,\n",
       "  fitType = \"parametric\")\n",
       "\\end{verbatim}\n",
       "\\end{Usage}\n",
       "%\n",
       "\\begin{Arguments}\n",
       "\\begin{ldescription}\n",
       "\\item[\\code{object}] a DESeqDataSet, or matrix of counts\n",
       "\n",
       "\\item[\\code{blind}] logical, whether to blind the transformation to the experimental\n",
       "design. blind=TRUE should be used for comparing samples in an manner unbiased by\n",
       "prior information on samples, for example to perform sample QA (quality assurance).\n",
       "blind=FALSE should be used for transforming data for downstream analysis,\n",
       "where the full use of the design information should be made.\n",
       "blind=FALSE will skip re-estimation of the dispersion trend, if this has already been calculated.\n",
       "If many of genes have large differences in counts due to\n",
       "the experimental design, it is important to set blind=FALSE for downstream\n",
       "analysis.\n",
       "\n",
       "\\item[\\code{intercept}] by default, this is not provided and calculated automatically.\n",
       "if provided, this should be a vector as long as the number of rows of object,\n",
       "which is log2 of the mean normalized counts from a previous dataset.\n",
       "this will enforce the intercept for the GLM, allowing for a \"frozen\" rlog\n",
       "transformation based on a previous dataset.\n",
       "You will also need to provide \\code{mcols(object)\\$dispFit}.\n",
       "\n",
       "\\item[\\code{betaPriorVar}] a single value, the variance of the prior on the sample\n",
       "betas, which if missing is estimated from the data\n",
       "\n",
       "\\item[\\code{fitType}] in case dispersions have not yet been estimated for \\code{object},\n",
       "this parameter is passed on to \\code{\\LinkA{estimateDispersions}{estimateDispersions}} (options described there).\n",
       "\\end{ldescription}\n",
       "\\end{Arguments}\n",
       "%\n",
       "\\begin{Details}\\relax\n",
       "Note that neither rlog transformation nor the VST are used by the\n",
       "differential expression estimation in \\code{\\LinkA{DESeq}{DESeq}}, which always\n",
       "occurs on the raw count data, through generalized linear modeling which\n",
       "incorporates knowledge of the variance-mean dependence. The rlog transformation\n",
       "and VST are offered as separate functionality which can be used for visualization,\n",
       "clustering or other machine learning tasks. See the transformation section of the\n",
       "vignette for more details, including a statement on timing. If \\code{rlog}\n",
       "is run on data with number of samples in [30-49] it will print a message\n",
       "that it may take a few minutes, if the number of samples is 50 or larger, it\n",
       "will print a message that it may take a \"long time\", and in both cases, it\n",
       "will mention that the \\code{\\LinkA{vst}{vst}} is a much faster transformation.\n",
       "\n",
       "The transformation does not require that one has already estimated size factors\n",
       "and dispersions.\n",
       "\n",
       "The regularization is on the log fold changes of the count for each sample\n",
       "over an intercept, for each gene. As nearby count values for low counts genes\n",
       "are almost as likely as the observed count, the rlog shrinkage is greater for low counts.\n",
       "For high counts, the rlog shrinkage has a much weaker effect.\n",
       "The fitted dispersions are used rather than the MAP dispersions\n",
       "(so similar to the \\code{\\LinkA{varianceStabilizingTransformation}{varianceStabilizingTransformation}}).\n",
       "\n",
       "The prior variance for the shrinkag of log fold changes is calculated as follows: \n",
       "a matrix is constructed of the logarithm of the counts plus a pseudocount of 0.5,\n",
       "the log of the row means is then subtracted, leaving an estimate of\n",
       "the log fold changes per sample over the fitted value using only an intercept.\n",
       "The prior variance is then calculated by matching the upper quantiles of the observed \n",
       "log fold change estimates with an upper quantile of the normal distribution.\n",
       "A GLM fit is then calculated using this prior. It is also possible to supply the variance of the prior.\n",
       "See the vignette for an example of the use and a comparison with \\code{varianceStabilizingTransformation}.\n",
       "\n",
       "The transformed values, rlog(K), are equal to\n",
       "\\eqn{rlog(K_{ij}) = \\log_2(q_{ij}) = \\beta_{i0} + \\beta_{ij}}{},\n",
       "with formula terms defined in \\code{\\LinkA{DESeq}{DESeq}}.\n",
       "\n",
       "The parameters of the rlog transformation from a previous dataset\n",
       "can be frozen and reapplied to new samples. See the 'Data quality assessment'\n",
       "section of the vignette for strategies to see if new samples are\n",
       "sufficiently similar to previous datasets. \n",
       "The frozen rlog is accomplished by saving the dispersion function,\n",
       "beta prior variance and the intercept from a previous dataset,\n",
       "and running \\code{rlog} with 'blind' set to FALSE\n",
       "(see example below).\n",
       "\\end{Details}\n",
       "%\n",
       "\\begin{Value}\n",
       "a \\code{\\LinkA{DESeqTransform}{DESeqTransform}} if a \\code{DESeqDataSet} was provided,\n",
       "or a matrix if a count matrix was provided as input.\n",
       "Note that for \\code{\\LinkA{DESeqTransform}{DESeqTransform}} output, the matrix of\n",
       "transformed values is stored in \\code{assay(rld)}.\n",
       "To avoid returning matrices with NA values, in the case of a row\n",
       "of all zeros, the rlog transformation returns zeros\n",
       "(essentially adding a pseudocount of 1 only to these rows).\n",
       "\\end{Value}\n",
       "%\n",
       "\\begin{References}\\relax\n",
       "Reference for regularized logarithm (rlog):\n",
       "\n",
       "Michael I Love, Wolfgang Huber, Simon Anders: Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2. Genome Biology 2014, 15:550. \\url{http://dx.doi.org/10.1186/s13059-014-0550-8}\n",
       "\\end{References}\n",
       "%\n",
       "\\begin{SeeAlso}\\relax\n",
       "\\code{\\LinkA{plotPCA}{plotPCA}}, \\code{\\LinkA{varianceStabilizingTransformation}{varianceStabilizingTransformation}}, \\code{\\LinkA{normTransform}{normTransform}}\n",
       "\\end{SeeAlso}\n",
       "%\n",
       "\\begin{Examples}\n",
       "\\begin{ExampleCode}\n",
       "\n",
       "dds <- makeExampleDESeqDataSet(m=6,betaSD=1)\n",
       "rld <- rlog(dds)\n",
       "dists <- dist(t(assay(rld)))\n",
       "# plot(hclust(dists))\n",
       "\n",
       "\\end{ExampleCode}\n",
       "\\end{Examples}"
      ],
      "text/plain": [
       "rlog                  package:DESeq2                   R Documentation\n",
       "\n",
       "_\bA_\bp_\bp_\bl_\by _\ba '_\br_\be_\bg_\bu_\bl_\ba_\br_\bi_\bz_\be_\bd _\bl_\bo_\bg' _\bt_\br_\ba_\bn_\bs_\bf_\bo_\br_\bm_\ba_\bt_\bi_\bo_\bn\n",
       "\n",
       "_\bD_\be_\bs_\bc_\br_\bi_\bp_\bt_\bi_\bo_\bn:\n",
       "\n",
       "     This function transforms the count data to the log2 scale in a way\n",
       "     which minimizes differences between samples for rows with small\n",
       "     counts, and which normalizes with respect to library size. The\n",
       "     rlog transformation produces a similar variance stabilizing effect\n",
       "     as ‘varianceStabilizingTransformation’, though ‘rlog’ is more\n",
       "     robust in the case when the size factors vary widely. The\n",
       "     transformation is useful when checking for outliers or as input\n",
       "     for machine learning techniques such as clustering or linear\n",
       "     discriminant analysis. ‘rlog’ takes as input a ‘DESeqDataSet’ and\n",
       "     returns a ‘RangedSummarizedExperiment’ object.\n",
       "\n",
       "_\bU_\bs_\ba_\bg_\be:\n",
       "\n",
       "     rlog(object, blind = TRUE, intercept, betaPriorVar,\n",
       "       fitType = \"parametric\")\n",
       "     \n",
       "     rlogTransformation(object, blind = TRUE, intercept, betaPriorVar,\n",
       "       fitType = \"parametric\")\n",
       "     \n",
       "_\bA_\br_\bg_\bu_\bm_\be_\bn_\bt_\bs:\n",
       "\n",
       "  object: a DESeqDataSet, or matrix of counts\n",
       "\n",
       "   blind: logical, whether to blind the transformation to the\n",
       "          experimental design. blind=TRUE should be used for comparing\n",
       "          samples in an manner unbiased by prior information on\n",
       "          samples, for example to perform sample QA (quality\n",
       "          assurance). blind=FALSE should be used for transforming data\n",
       "          for downstream analysis, where the full use of the design\n",
       "          information should be made. blind=FALSE will skip\n",
       "          re-estimation of the dispersion trend, if this has already\n",
       "          been calculated. If many of genes have large differences in\n",
       "          counts due to the experimental design, it is important to set\n",
       "          blind=FALSE for downstream analysis.\n",
       "\n",
       "intercept: by default, this is not provided and calculated\n",
       "          automatically. if provided, this should be a vector as long\n",
       "          as the number of rows of object, which is log2 of the mean\n",
       "          normalized counts from a previous dataset. this will enforce\n",
       "          the intercept for the GLM, allowing for a \"frozen\" rlog\n",
       "          transformation based on a previous dataset. You will also\n",
       "          need to provide ‘mcols(object)$dispFit’.\n",
       "\n",
       "betaPriorVar: a single value, the variance of the prior on the sample\n",
       "          betas, which if missing is estimated from the data\n",
       "\n",
       " fitType: in case dispersions have not yet been estimated for ‘object’,\n",
       "          this parameter is passed on to ‘estimateDispersions’ (options\n",
       "          described there).\n",
       "\n",
       "_\bD_\be_\bt_\ba_\bi_\bl_\bs:\n",
       "\n",
       "     Note that neither rlog transformation nor the VST are used by the\n",
       "     differential expression estimation in ‘DESeq’, which always occurs\n",
       "     on the raw count data, through generalized linear modeling which\n",
       "     incorporates knowledge of the variance-mean dependence. The rlog\n",
       "     transformation and VST are offered as separate functionality which\n",
       "     can be used for visualization, clustering or other machine\n",
       "     learning tasks. See the transformation section of the vignette for\n",
       "     more details, including a statement on timing. If ‘rlog’ is run on\n",
       "     data with number of samples in [30-49] it will print a message\n",
       "     that it may take a few minutes, if the number of samples is 50 or\n",
       "     larger, it will print a message that it may take a \"long time\",\n",
       "     and in both cases, it will mention that the ‘vst’ is a much faster\n",
       "     transformation.\n",
       "\n",
       "     The transformation does not require that one has already estimated\n",
       "     size factors and dispersions.\n",
       "\n",
       "     The regularization is on the log fold changes of the count for\n",
       "     each sample over an intercept, for each gene. As nearby count\n",
       "     values for low counts genes are almost as likely as the observed\n",
       "     count, the rlog shrinkage is greater for low counts. For high\n",
       "     counts, the rlog shrinkage has a much weaker effect. The fitted\n",
       "     dispersions are used rather than the MAP dispersions (so similar\n",
       "     to the ‘varianceStabilizingTransformation’).\n",
       "\n",
       "     The prior variance for the shrinkag of log fold changes is\n",
       "     calculated as follows: a matrix is constructed of the logarithm of\n",
       "     the counts plus a pseudocount of 0.5, the log of the row means is\n",
       "     then subtracted, leaving an estimate of the log fold changes per\n",
       "     sample over the fitted value using only an intercept. The prior\n",
       "     variance is then calculated by matching the upper quantiles of the\n",
       "     observed log fold change estimates with an upper quantile of the\n",
       "     normal distribution. A GLM fit is then calculated using this\n",
       "     prior. It is also possible to supply the variance of the prior.\n",
       "     See the vignette for an example of the use and a comparison with\n",
       "     ‘varianceStabilizingTransformation’.\n",
       "\n",
       "     The transformed values, rlog(K), are equal to rlog(K_ij) =\n",
       "     log2(q_ij) = beta_i0 + beta_ij, with formula terms defined in\n",
       "     ‘DESeq’.\n",
       "\n",
       "     The parameters of the rlog transformation from a previous dataset\n",
       "     can be frozen and reapplied to new samples. See the 'Data quality\n",
       "     assessment' section of the vignette for strategies to see if new\n",
       "     samples are sufficiently similar to previous datasets.  The frozen\n",
       "     rlog is accomplished by saving the dispersion function, beta prior\n",
       "     variance and the intercept from a previous dataset, and running\n",
       "     ‘rlog’ with 'blind' set to FALSE (see example below).\n",
       "\n",
       "_\bV_\ba_\bl_\bu_\be:\n",
       "\n",
       "     a ‘DESeqTransform’ if a ‘DESeqDataSet’ was provided, or a matrix\n",
       "     if a count matrix was provided as input. Note that for\n",
       "     ‘DESeqTransform’ output, the matrix of transformed values is\n",
       "     stored in ‘assay(rld)’. To avoid returning matrices with NA\n",
       "     values, in the case of a row of all zeros, the rlog transformation\n",
       "     returns zeros (essentially adding a pseudocount of 1 only to these\n",
       "     rows).\n",
       "\n",
       "_\bR_\be_\bf_\be_\br_\be_\bn_\bc_\be_\bs:\n",
       "\n",
       "     Reference for regularized logarithm (rlog):\n",
       "\n",
       "     Michael I Love, Wolfgang Huber, Simon Anders: Moderated estimation\n",
       "     of fold change and dispersion for RNA-seq data with DESeq2. Genome\n",
       "     Biology 2014, 15:550. <URL:\n",
       "     http://dx.doi.org/10.1186/s13059-014-0550-8>\n",
       "\n",
       "_\bS_\be_\be _\bA_\bl_\bs_\bo:\n",
       "\n",
       "     ‘plotPCA’, ‘varianceStabilizingTransformation’, ‘normTransform’\n",
       "\n",
       "_\bE_\bx_\ba_\bm_\bp_\bl_\be_\bs:\n",
       "\n",
       "     dds <- makeExampleDESeqDataSet(m=6,betaSD=1)\n",
       "     rld <- rlog(dds)\n",
       "     dists <- dist(t(assay(rld)))\n",
       "     # plot(hclust(dists))\n",
       "     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "?rlog"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "counts <- counts[rownames(model),]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "converting counts to integer mode\n",
      "  it appears that the last variable in the design formula, 'Sample_Group',\n",
      "  has a factor level, 'Control', which is not the reference level. we recommend\n",
      "  to use factor(...,levels=...) or relevel() to set this as the reference level\n",
      "  before proceeding. for more information, please see the 'Note on factor levels'\n",
      "  in vignette('DESeq2').\n"
     ]
    }
   ],
   "source": [
    "de <- DESeqDataSetFromMatrix(as.matrix(round(counts)), colData = sample.info, design=~Sample_Group)"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "dat <- rlog(de, blind=FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "class: DESeqTransform \n",
       "dim: 6 10 \n",
       "metadata(1): version\n",
       "assays(1): ''\n",
       "rownames(6): ENSG00000091972 ENSG00000187601 ... ENSG00000003147\n",
       "  ENSG00000149257\n",
       "rowData names(7): baseMean baseVar ... dispFit rlogIntercept\n",
       "colnames(10): 6a 8a ... B5 B6\n",
       "colData names(6): Sample_Biosource Concentration ... Sample_Group\n",
       "  sizeFactor"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(dat)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0gAAANICAIAAAByhViMAAAACXBIWXMAABJ0AAASdAHeZh94\nAAAgAElEQVR4nOzde3wTVf7/8TO5N01aeqVcalvkbpHrIiBeUCuCioCuCyogy4osgrJyEb67\nq66uoIsKKHgB1PUK6yqIyyK4eEUFRbcIiCh3W+60tGmbprnN74+s/dVS0jQ0Sef09fyDRzJz\nmHymmUzeOTNzRlFVVQAAAED7dLEuAAAAAI2DYAcAACAJgh0AAIAkCHYAAACSINgBAABIgmAH\nAAAgCYIdAACAJAh2AAAAkiDYAQAASIJgBwAAIAmCHQAAgCQIdgAAAJIg2AEAAEiCYAcAACAJ\ngh0AAIAkCHYAAACSINgBAABIgmAHAAAgCYIdAACAJAh2AAAAkiDYAQAASIJgBwAAIAmCHQAA\ngCQIdgAAAJIg2AEAAEiCYAcAACAJgh0AAIAkCHYAAACSINgBAABIgmAHAAAgCYIdAACAJAh2\nAAAAkiDYAQAASIJgBwAAIAmCHQAAgCQIdgAAAJIg2AEAAEiCYAcAACAJgh0AAIAkCHYAAACS\nINgBAABIgmAHAAAgCYIdAACAJAh2AAAAkiDYAQAASIJgBwAAIAmCHQAAgCQIdgAAAJIg2AEA\nAEiCYAcAACAJgh0AAIAkCHYAAACSINgBAABIgmAHAAAgCYIdAACAJAh2AAAAkiDYAQAASIJg\nBwAAIAmCHQAAgCQIdgAAAJIg2AEAAEiCYAcAACAJgh0AAIAkCHYAAACSINgBAABIgmAHAAAg\nCYIdAACAJAh2AAAAkiDYAQAASIJgBwAAIAmCHQAAgCQIdgAAAJIg2AEAAEiCYAcAACAJgh0A\nAIAkCHYAAACSINgBAABIgmAHAAAgCYIdAACAJAh2AAAAkiDYAQAASIJgBwAAIAmCHQAAgCQI\ndgAAAJIg2AEAAEiCYAcAACAJgh0AAIAkCHYAAACSINgBAABIgmAHAAAgCYIdAACAJAh2AAAA\nkiDYAQAASIJgBwAAIAmCHQAAgCQIdgAAAJIg2AEAAEiCYAcAACAJgh0AAIAkCHYAAACSINgB\nAABIgmAHAAAgCYIdAACAJAh2AAAAkiDYAQAASIJgBwAAIAmCHQAAgCQIdgAAAJIg2AEAAEiC\nYAcAACAJgh0AAIAkCHYAAACSINgBAABIgmAHAAAgCYIdAACAJAh2AAAAkiDYAQAASIJgBwAA\nIAmCHQAAgCQIdgAAAJIg2AEAAEiCYAcAACAJgh0AAIAkCHYAAACSINgBAABIgmAHAAAgCYId\nAACAJAh2AAAAkiDYAQAASIJgBwAAIAmCHQAAgCQIdgAAAJIg2AEAAEiCYAcAACAJgh0AAIAk\nCHYAAACSINgBAABIgmAHAAAgCYIdAACAJAh2AAAAkiDYAQAASIJgBwAAIAmCHQAAgCQIdgAA\nAJIg2AEAAEiCYAcAACAJgh0AAIAkCHYAAACSINgBAABIgmAHAAAgCYIdAACAJAh2AAAAkiDY\nAQAASIJgBwAAIAmCHQAAgCQIdgAAAJIg2AEAAEiCYAcAACAJgh0AAIAkCHYAAACSINgBAABI\ngmAHAAAgCYIdAACAJAh2AAAAkiDYAQAASIJgBwAAIAmCHQAAgCQIdgAAAJIg2AEAAEiCYAcA\nACAJgh0AAIAkCHYAAACSINgBAABIgmAHAAAgCYIdAACAJAh2AAAAkiDYAQAASIJgBwAAIAmC\nHQAAgCQIdgAAAJIg2AEAAEiCYAcAACAJgh0AAIAkCHYAAACSINgBAABIgmAHAAAgCYIdAACA\nJAh2AAAAkiDYAQAASIJgBwAAIAmCHQAAgCQIdgAAAJIg2AEAAEiCYAcAACAJgh0AAIAkCHYA\nAACSINgBAABIgmAHAAAgCYIdAACAJAh2AAAAkiDYAQAASIJgBwAAIAmCHQAAgCQIdgAAAJIg\n2AEAAEiCYAcAACAJgh0AAIAkCHYAAACSINgBAABIgmAHAAAgCYIdAACAJAh2AAAAkiDYAQAA\nSIJgBwAAIAmCHQAAgCQIdgAAAJIg2AEAAEiCYAcAACAJgh0AAIAkCHYAAACSINgBAABIgmAH\nAAAgCYIdAACAJAh2AAAAkiDYAQAASIJgBwAAIAmCHQAAgCQIdgAAAJIg2AEAAEiCYAcAACAJ\ngh0AAIAkCHYAAACSINgBAABIgmAHAAAgCYIdAACAJAh2AAAAkiDYAQAASIJgBwAAIAmCHQAA\ngCQIdgAAAJIg2AEAAEiCYAcAACAJgh0AAIAkCHYAAACSINgBAABIgmAHAAAgCYIdAACAJAh2\nAAAAkiDYAQAASIJgBwAAIAmCHQAAgCQIdgAAAJIg2AEAAEiCYAcAACAJgh0AAIAkCHYAAACS\nINgBAABIgmAHAAAgCYIdAACAJAh2AAAAkiDYAQAASIJgBwAAIAmCHQAAgCQIdgAAAJIg2AEA\nAEiCYAcAACAJgh0AAIAkCHYAAACSINgBAABIgmAHAAAgCYIdAACAJAh2AAAAkiDYAQAASIJg\nBwAAIAmCHQAAgCQIdgAAAJIg2AEAAEiCYAcAACAJQ6wLqEdFRUVlZWWsqwiToigpKSlut9vh\ncMS6lvAlJSWVlJSoqhrrQsJktVqtVqvD4XC73bGuJUxGo9FisZSVlcW6kPAlJycLIYqLi2Nd\nSPjsdrvL5fJ4PLEuJEwmkykhIcHpdDqdzljXEiZFURITE0tKSmJdSPgSEhJMJlNRUZGm96h+\nv9/lctU5NzU1Ncr14Ez02AEAAEiCYAcAACAJgh0AAIAkmvo5dgAA4BwplU7jN18a9v6oKylW\nDEZfSqqnS66nW0+h18e6NDQygh0AADIzHNhnWfu2Un3Fg8ejL/xJX/iTKX9r5Yjf+Fskx7Q6\nNDIOxQIAIC194U+W1f9Q6rqOVXfqpPUfryrOiuhXhcgh2AEAICm/37L+X4rPe7b5iqPU/MkH\n0awIkUawAwBAToZD+3Wni4K3MX6/o87+PGgUwQ4AADnpC36qv5HPpz9aGPlaECUEOwAA5KRU\nhnb+XAWn2cmDYAcAgJzUOGtI7ayhNYMWEOwAAJCTr01m/Y10Ol+rtpGvBVFCsAMAQE7e7PP9\niS2Ct/F0vkCNi4tOPYgCgh0AAJLS611XX6ee/fYSaryt6rKrolkRIo1gBwCAtHzZ7VzXjVSN\nxjNn+VskOW++TbXZo18VIodbigEAIDNvxy4VrdqYtm427P1BV+ZQFcWfkurtkuvp1Vc11BH4\noGkEOwAAJKfaE6quGFx1xWDh9wsdB+tkxrsLAECzQaqTHW8wAACAJAh2AAAAkiDYAQAASKKp\nXzyh1+sNhqZe5NkoihL4V7urEGAwGFRVjXUVYdLpdELjG5Jer5dgKxJCaHoVFEXR6/Xa/SDo\n9XohhE6n0+i7oDu037Bzu+/kcZu7Sk1I9OW09/Xoo5rNsa6rYQJfClrfo55td6TdlZKM0sTf\nCa/X6/P5Yl1FmBRFMZlMfr/f4/HEupbwmUwmt9sd6yrCF4h0Ho/H7/fHupYw6XQ6nU7n9Xpj\nXUj4TCaTEELTG5LRaPT5fJreioxGoyb3qFVVYvVKsWtn7enWePHrW8T5HWNRU5iMRqNOp6uq\nqop1IeEL/EKocyvy+XxW7jnbBDT1YFdRUVFZWRnrKsKkKEpKSorb7XY4HLGuJXxJSUklJSVN\nfDsJwmq1Wq1Wh8Oh3VRhNBotFktZWVmsCwlfcnKyEKK4uDjWhYTPbre7XC7t/kgzmUwJCQlO\np9PpdMa6lobw+az/fE1fcKjOmapeX/nrW32Z2dGtKXwJCQkmk6moqEjTe1S/3+9yueqcm5qa\nGuV6cCbOsQMANFGm/351tlQnhFB8vrj33lV8Gu7MBhodwQ4A0CSpqumbLcGbKKUl+h92Racc\nQBMIdgCApkhXXKSEcAaC4dCBKBQDaAXBDgDQFCkV5SE1Kw+pGdBMEOwAAE1SaKOZqBZLpAsB\nNIRgBwBoivypaarRVH+z1m2jUAygFQQ7AEBTpOoN3gsurKeN0ejpfEF06gE0gWAHAGiiqi6+\nTLXbgzRwX3KFGm+LWj1A00ewAwA0Uao1vvKmW9XEFnXOrep/ibv3RVEuCWjiNHnTQABAM+FL\nTa8YO9G49QvTrh2Ko1QIIfR673k57v6X+Npkxro6oMkh2AEAmjTVYnFfcoXn0isTDAZH0Sk1\n3ib0+lgXBTRRBDsAgDYo8fGqZm/XC0QH59gBAABIgmAHAAAgCYJd43CpaoXfH+sqAABAs8Y5\nduek2Od7+mTxGkdZgdsjhEjW64cm2O5JS8k2GWNdGgAAaHbosQvfVqfr4j0HFp8qDqQ6IUSx\nz/fa6dJL9hxYXVoW29oAAEAzRLAL074q9+hDhae8vjNnuVR1SuHRT8orol8VAABozgh2YfrT\nsROlvjpSXYBbVWccOe5R1WiWBAAAmjmCXTgOezwflNXTIXfQ7fmUTjsAABBFBLtwfFPpCqUv\nbqvTFfFSAAAAftaAYOdyubZu3bp69epTp04JIbxeb8SqaupK6jq17kynz36sFgAAoNGFGuye\neOKJ9PT0vn37jhw5cu/evUKIBx54YPz48c0z3qUZQhomJt3A3QwBAED0hBTsli1bNmPGjEGD\nBj333HPVEzt16vTaa68tWLAgYrU1Xf2scQZFqbfZwHhrFIoBAAAICCnYLV68eNKkSWvWrBk3\nblz1xLFjx86cOXP58uURq63pSjLof51oD96ml9XSl2AHAACiKKRg9+OPP954441nTr/88ssP\nHDjQ2CVpwwOt0jONZz0gG6/TLWydUX+fHgAAQOMJ6VyxhIQEl6uOCzxLS0vj4uIa+pJ33333\nwYMHq59aLJY333yzoQuJuRS9fk27rLGHCne6qmrNam00vJjZuovFHJPCAABAsxVSsLvwwgsf\nf/zxK6+8UqlxYllxcfFDDz3Ur1+/hr5keXn5xIkTq/+jTqfVIVcyjYb/nJ/1VoljjaPshyq3\nR1XbmUxDEmxjkxKtml0pAACgXSEFuz/+8Y9XXXXVhRdeeO211wohli1b9txzz61evbqysrLm\n5RQhKisry8jISE1NbXCxTY9BUUYlJY5KSox1IQAAAEJRQ7vt1QcffDBz5sz8/PzqKX379v3b\n3/522WWXNej1PB7PjTfemJeXt2/fvrKysvbt248dO7ZNmzbVDfbt21dUVBR4bDAYunbtWlVV\n+1inViiKkpCQ4PV6Kyo0fAsKu91eXl4e4nbSBJnNZovFUlFRod2heQwGg8lkcjqdsS4kfHa7\nXQhRVlYW60LCZ7Va3W63drcio9FotVpdLpem96jx8fHl5eWxLiR88fHxBoPB4XBod49qsVj8\nfr/b7T5zlqqqLVq0iH5JqCXUYBdw4sSJI0eOCCGysrKSkpLCeL3S0tKpU6f27Nkz0Pm3YsWK\nffv2Pfvss/Hx8YEGf/rTn9avXx94nJSU9J///CeMVwEAANHk8XiMRmOsq0DIwe7YsWP//Oc/\np06dGnh68uTJZ599dtKkSenp6efy8pWVlePGjbvjjjvy8vICU95///0ffvgh8DguLm7cuHHa\n/YmsKIrFYvH5fHX+uNEKi8VS56UzWmEwGIxGo9vt9mn2RiA6nc5gMGh9KxJCaHpDMplMXq/X\n7/fHupAw6fV6k8nk8Xi0u0cVQpjNZu32OAohTCaTXq93uVza7bEzGAyqqta5O/X5fDabLfol\noZaQgt0PP/xw+eWXFxcXV3+iDh06lJ2dnZGR8fnnn7dr1+5cKrjrrrsGDhw4evToOudWVFRU\nVlaey/JjSFGUlJQUt9vtcDhiXUv4kpKSSkpKtLsbslqtVqvV4XBoNxgZjUaLxaLp45jJyclC\niOLi4lgXEj673e5yuTweT6wLCZPJZEpISHA6ndo9pq8oSmJiYklJSawLCV9CQoLJZCoqKtL0\nHtXv95/tR5ocZ89rXUgXb86ePdtms3322WfVU7Kysnbt2mWz2WbOnNmg1zt06NDixYurfzK6\nXK6TJ09mZGQ0aCEAAAA4U0hXxX7++eePPfbYr371q5oTu3TpMnPmzBkzZjTo9ZKTkzdv3uz1\nekeNGuXz+V555RWbzTZgwIAGLQQAAABnCinYlZeXm0ymM6fbbLaGnrdkt9sffvjhl156adq0\naUajsVOnTvPmzTObGcsXAADgXIUU7Hr27Pnqq6+OGjVKr9dXTywrK1u4cGHPnj0b+pLt2rV7\n+OGHG/q/AAAAEFxIwe7+++8fMmRIx44dhwwZkpaW5vf7CwoK1q5dW1RUtG7dukiXCAAAgFCE\nFOwGDx68YcOGOXPmLFmypHrihRde+Pe//33w4MERqw0AAAANEFKwE0Lk5eXl5eUVFRUdOXJE\nr9dnZmYGhpIHAABAExFqsAtISUlJSUmJUCkAAAA4FyGNY3fixInbb7+9TZs2er1eOUOkSwQA\nAEAoQuqxmzJlyurVqy+77LK8vDyDoWGdfAAAAIiOkFLahx9++NZbb91www2RrgYAAABhC+lQ\nbGVlJTeHAAAAaOJCCna9e/f+7rvvIl0KAAAAzkVIwW7BggX33Xff5s2bI10NAAAAwhbSOXb3\n3HPP0aNHBwwYYLVa09LSas09ePBg49cFAACABgop2Ol0uo4dO3bs2DHS1QAAACBsIQW7Tz/9\ntM7p5eXlR48ebdR6AAAAEKaQzrE7my+//LJfv36NVQoAAADORaijDf/73/9esWLFTz/95Pf7\nA1N8Pt93331nNpsjVhsAAAAaIKRgt3LlytGjRxsMhoyMjMLCwtatWxcXF7tcrkGDBs2YMSPS\nJQIAACAUIR2Kffzxx6+55pri4uKCggK9Xr9hw4aysrKnnnpKVdVLLrkk0iUCAAAgFCEFux9/\n/HHKlCl2uz3wVFVVg8EwderUHj16zJkzJ5LlAQAAIFQhBTuPx6PX6wOP4+PjS0pKAo9vvPHG\n1atXR6o0AAAANERIwa5Lly4vvPCC2+0WQmRmZm7YsCEwvbi4uLS0NILVAQCAZszv9z/44IOZ\nmZkWi6V3797/+c9/pk6dajKZAnMHDhx46aWXrl27NjMzs/qm9u+9996ll15qt9vj4uJyc3Of\nfPJJVVUDs3r06NGjR4+ayx8+fHhqamrgce/evfv37//hhx/27dvXarUmJyf/9re/1VzOCeni\niXvvvXfMmDGnT5/euHHjyJEj586de+LEibZt2y5durR79+6RLhEAADRPjz766F/+8pebb755\nwoQJBQUF48aNy8zMrA52ZrP51KlTM2fOnDNnTlZWlhDinXfeGTly5ODBg1977TWbzbZu3brp\n06cfO3bsb3/7W72vZTab9+7de9999y1atKhjx47r16//7W9/W1JSsmrVqsiuZKMKKdjddttt\nBoMhcOuw2bNnb9myZdmyZUKIzMzMRYsWRbQ+AADQPKmq+tRTT+Xm5q5cuVJRFCFEbm5uv379\n4uPjAw0URdm+ffuqVatGjBgRmDJnzpzMzMw1a9YEwt+VV165f//+hQsX3nfffSkpKcFfTqfT\nnTx58u233w50/t12220ff/zxCy+8UFBQkJmZGcH1bFShDlA8atSo2bNnCyGsVuv777+/Z8+e\nXbt27du3r0+fPpEsDwAANFPHjh07fvx4Xl5eINUJIS666KLc3NyabUwm03XXXRd4fOTIkd27\ndw8dOrS6S08Icf3113s8ni1btoTyivHx8QMHDqx+eumllwohdu7ceY4rEk2hDlBcS/v27Ru3\nDgAAgJqOHz8uhGjVqlXNiZ06dTpw4ED109TUVKPRGHh8+PBhIUSbNm1qtg/89yNHjoTyii1b\ntqwOkUKIQCdfoAytCBbsOnfuPG7cuDlz5nTu3DlIs927dzd2VQAAoLmrqqoSQuh0vzi6WDN4\nCSGqU131rOpbZAUErpyotZAQeb3esP9vrAQLdi1atIiLiws8iFY9tRmNxuqLWTQnsIXpdDqL\nxRLrWsKnKIrFYtHuu2AwGIQQJpNJW5/MmvR6vV6v1/pWJITQ9Cro9XqTyVQ98JPmBD4IBoNB\nu++Coiha350Gth+t71HPVnytONUokpOTxRkdZj/88MPZ2rdt21b83G9XLfA0MEun03k8nppz\njx07VvPp0aNHfT5f9Sc98NItW7Y8h5WItmDBrvqAdIhHpiOkVjbXkEDliqJodxUCtF5/gHbX\nonpDinUh50rrqyDBZ1lo+V3gg9BEnO2DEIn1ysnJSUxMfO+996qvad26deuOHTuqL56oJSMj\nIzc3d+3atS6Xq/o3wKpVq6xWa//+/YUQSUlJ27dvV1U1UO2JEye2b99utVqrl1BZWfn+++8P\nGTIk8PS9994zm819+/Zt9FWLnJDOsRswYMCf/vSnoUOHRrqaM3k8nsrKyui/bqNQFMVqtfp8\nPu2ughDCYrFUVlZq9/eloigmk8ntdgcGYtQio9GoKIqmt6JA37+mV8FgMFRVVdX6ra8hJpPJ\nYrF4vV7tvguBz7J26xdCGI1GvV6v9T2q3+93uVx1zq2+Q1VjMRgMEyZMePLJJ8ePHz969OiD\nBw/Omzfv4osv3rZt29n+y2OPPXb99dffcMMNd911l8lkevfdd9evXz9v3ryEhAQhxLBhwz78\n8MPHHnts/PjxR44cmT59ert27Wp22mVmZk6bNu3QoUPt27ffsGHDO++8M3bs2KSkpMZdr4gK\n6eBUQUEBJ9IBAIAomzt37tSpU9euXTty5MjXXnvtH//4R1ZWVpDewaFDh65fv97pdN5yyy3D\nhw/fsmXLiy++GBjWQwjx+9///t577128eHFWVtb48ePvvffeyy+/vObP/vj4+DfeeGPFihU3\n3HDDCy+8cMcddyxZsiTiK9molFB+N7z77ruzZ8+eO3futddeW/MsxSioqKjQ7u8zRVFSUlLc\nbrfD4Yh1LeFLSkoqKSnR7u9Lq9VqtVodDoeme+wsFktZWVmsCwlf4ESZ4uLiWBcSPrvd7nK5\nNN1jl5CQ4HQ6nU5nrGsJk6IoiYmJ1fe01KKEhASTyVRUVKTpPWqQHrvqWzhE1FVXXbVr164Q\nr3JtkIEDB546dUrrPVkhHYp9/PHHDQbDiBEjTCZTzeuKAwIDFwMAADSuhQsXbtq06R//+Efg\nAqCSkpKvv/66+u5hOFNIwc7v96elpV155ZWRrgYAAKBaSkpK4MYSd9xxh8vlWrhwocPhmD59\neqzrarpCCnafffZZndPLy8uPHj3aqPUAAAD8z5gxY4QQCxYsuOWWW1RV7dGjx9q1a+lpCiLM\nO08EfPnllzfffHNRUVFjVQMAAFDTmDFjAvEu0s7WjaUtoQa7f//73ytWrPjpp5+qRyD0+Xzf\nffed2WyOWG0AAABogJCC3cqVK0ePHm0wGDIyMgoLC1u3bl1cXOxyuQYNGjRjxoxIlwgAAIBQ\nhDSO3eOPP37NNdcUFxcXFBTo9foNGzaUlZU99dRTqqpecsklkS4RAAAAoQgp2P34449Tpkyp\nHlFaVVWDwTB16tQePXrMmTMnkuUBAAAgVCEFO4/HU31D3Pj4+OrxIW+88cbVq1dHqjQAAAA0\nREjBrkuXLi+88EJg4P7MzMwNGzYEphcXF5eWlkawuhg57fWtdZQvKypZebr0e1dVrMsBAAAI\nSUgXT9x7771jxow5ffr0xo0bR44cOXfu3BMnTrRt23bp0qXdu3ePdInR5PD5Hzp+8o3TpZ4a\n93vpGWeZ2yq9jzUuhoUBAADUK6Rgd9tttxkMhsCtw2bPnr1ly5Zly5YJITIzMxctWhTR+qLp\nlNd3w4GffqyqfUfR/ErX8AMFz2a2vj7BFpPCAABonhr9vrqKojTuApuakIKdz+cbNWpU4LHV\nan3//ff37t3r8Xjat29f676xmjax8MiZqS6gSlXvKjzS5fzs9mZTlKsCAKDZKi8vb9wF2mw2\nubNdSOfYZWZmTp8+fdu2bdVT2rdv36VLF5lS3Uflzk3lziANKv3qoydORa0eAACAhgop2GVl\nZS1YsKBnz565ubmPPfZYQUFBpMuKHKffv7Gs4uXikjdLSnfVuDBiTamj3v/7fll5VWP3CQMA\nADSWkILd5s2bDx48OH/+fKvVOnv27KysrEGDBr344osOR/1hqOmoUtWHjp/svHvv6EOFM44c\nv6vw2GV7D16+9+DnFU4hxN6zHIStqdKvFrg9ka8UAAAgHCEFOyHEeeedN2PGjK+++urAgQOP\nPvpoeXn5hAkTWrZs+Zvf/Cai9TWWCr9/+IGCp08WV/pVUaPT7TtX1c0HC1eUOPyhLEUVPnrs\nAABAUxVqsKuWnZ09a9asrVu3rlq1qnXr1m+++WYkymp09x4+/rWz8n9PfnnSpFtVZxw+lqjT\n17sQk17JNMlzWiEAAJBMSFfFVvP5fJs2bXrrrbdWr1595MiR5OTkO+64I0KVNaLtla5VQU+h\nc6vqUW/9x1gvi7dadQ2OwgAAIKKUqirVbI51FU1CSMHO6/V+9NFHb7311jvvvHPixAmr1Xr9\n9dffcsstQ4YM0cSFsf9y1H+x9G6X+wKL+TtXlVBrd+kFmBRlVnpq4xcHAADCotuzW7d1s67g\nkPB4hMGgts3y9fqVv0uukHpAk+BCCnYtW7YsLi42GAx5eXm33HLLiBEj4uPjI11ZIzrb6HQ1\n+YQ6PT11zpHjx73eOhs8nJHWI87S2KUBAICG83r1/3pb/932mlOUg/sMB/f5O3bxjfiNeg4d\nT4WFhXPnzl23bt3Ro0eTkpL69u07c+bMSy65pBHKjryQDix27dp1yZIlR48eXbdu3W233aat\nVCeE8IZ2xUOqXv+f87Outte+vUSmyfjqeW1+m5IUgdIAAECDGda984tUV4Pux+/1a8K/AGD3\n7t09e/b89NNPn3zyyfz8/BUrViQmJg4aNOjtt98Oe5nRFFKw27Rp0+TJk1NTtXogMieEKx4U\nIXJMxlZGw+tZbT5rn/1Iq/S7UpPvS09dmd12S/uca7iZGAAATYNycJ9ue36QBrrdu3S7d4W3\n8MmTJ6elpX399dcjR47s2rXroEGDXn311VmzZu3YsSPQYOfOnVdffXVycnKLFuNh7SYAACAA\nSURBVC0GDx68d+9eIYTf71cUZcWKFYMHD+7atWtWVtbLL78caF9YWDhixAibzZaRkTF58mSn\n0ymEOHbs2KhRo1q3bh0fH3/ZZZf997//FUL4fD5FUZYvX56TkzN+/Pjw6m/YxRONory8fOnS\npdu3b/d4PJ06dZo0aVJ6enpEX3FIgu35otPB2/SIs2QY//fX6GQxd7JwDmZtxV7f6lLHV05X\nkc+XbtD3s8aNSEyw67maBAAQVfr8r+tto/vvV/7OXRu65JMnT3700UcvvfSSxfKLk6/mzp1b\n/fimm2666KKLCgoKfD7fb3/723Hjxn3++ec6nU6v1z/xxBPr1q1LT09/4YUXJk+efNNNN8XH\nx48cOTI7O3vPnj3l5eUjRoyYNWvW4sWLhw8fnp2dvWPHDqvV+sgjjwwZMuTgwYNxcXF6vf75\n559/++23O3To0NDiA2IQ7BYuXFheXv7AAw+YzeY33njjoYceeuqpp3SRvNr04njrwHjrZxXB\n7hg2p2Va5AqQwMoSxx+PHnf4/v94f/8sccw9furxNhnX0Z0JAIgiXeFP9TVRlYJDYSx5//79\nQojc3NwgbTZv3mw2m61WqxDilltuGTVqlKqqgfvPjhkzJtBXdeWVVzqdzoMHD3o8nq1bt65Y\nsaJVq1ZCiFdfffXIkSP//e9/v/zyy9WrV6ekpAghHnrooSVLlrz77ruBsYGHDx/eq1evMIoP\niHZ3y6lTp7Zu3Tpx4sScnJzWrVtPmjTp8OHD1d2bkfNcZqtM4xkp9udT76anpQyyWSNdg3Yt\nKzo9tfBozVQnhBCqKPL5xv90+J8lWroBCQBA61RnRX1NFMXjFj5feMv3nuUyyoD8/Pzrrrsu\nIyMjIyNjwoQJHo/H9/MLnXfeeYEHgQ6/ysrKvXv3KoqSk5MTmN6zZ89rr732xx9/FEK0bt1a\nURRFUfR6fUlJSSBTCiHat28fXtkB0Q52e/bsMRqN1Wtos9natm37ww8/RPp1WxoM75+ffV2i\n/RdTFZGi1z/dttXsllo9fTAKdlU4Hzh2oo4ZP19LPv3IsSOeYJ8BAAAaU3z9R4pUS5zQ13/r\ngVo6duyoKEp+fu0T+Hw+n6qqQoi9e/cOHTo0Ly/v4MGDx44d+/vf/16zmXLGMCuBKeovL+KM\ni4sTQlRWVqo1zJkzJzDXfG4D8jXsUGx+fv6aNWsKCwuFENnZ2YHzChu0BIfDYbfba655YmJi\naWlp9dNPP/30wIEDgccWi2XEiBENWn4QmUKstNt+dFX9p9RxqMpt1+u7WePyEmzxDX/jQxRY\nTb1eH3gLNerJnw67/cEuK670q8tLHfMy20StpAYxGAxCCJPJpI/YGx1per1e61tR4LOg6VXQ\n6/VmszmwOWlRYPs3GAzafRcURdHpdNqtX/z8LsTFxamavTul0WisPuxYi98f0r05G4Wa1U4p\n+aaeNtk5YSw5KSnp6quvfvTRR2+99daEhITq6ffff/+WLVs++OCDr7/+2uv1zpgxIzCO75Yt\nW4IvsH379qqqfv/994HDu1999dVXX311xRVXCCG2bdvWr1+/QLP9+/e3a9cujILP1ICd1NNP\nP33PPffk5uaed955qqp+/vnnDz744LJlyxp64UadG0S1999/f/369YHHSUlJv/nNb0wmU4OW\nH1zP+PieKcmNuMB66fV6zQ0QU9OG06X1tvmgrKKJr2Ot02C1SLuRoloT30jqJcFbYDKZGneP\nGn1a34qEEIFzs9TyMlHlUuLtQoN7pzq3Io+n/hs4NRb/r/rpvq0n2Pl+NSC8hT/99NMXX3xx\njx49Hnnkke7du588efLFF198880316xZI4TIzs72+Xxbtmzp27fvqlWrvvjiCyHEkSNHqg/C\n1tK9e/eLLrpo+vTpzz33nMfjufPOO/v37z9lypQrrrhi+vTpgXPvli9fPmPGjD179rRu3Tq8\nmmtqwH5q/vz569atu+aaa6qnrF69esaMGQ0Kdi1atHA4HDXzfmlpaVLS/x8ibtSoUZdffnng\nsclkqqqqcrvrH164aVIUxWazeb3eysrK+ls3SR5VPRbC3/9QpausrCwK9YTBZDKZzebKysrg\n50w0ZXq93mg0ulyuWBcSPpvNJoQoL6//HjBNlsViqXkmjeYE+uo0vUcVQlit1sBQERoVFxdn\n8PsrP9ig5G9VSkuEEEJR/K3a+C+62N+1myZulmAymVRVrTPD+f3+xMTE6JThz2jtu/gy/eef\nnLVBn35qVjg9dkKIDh06fPPNNw8//PB99913/PjxlJSUiy++ePPmzT169BBC9OvXb+bMmTfc\ncIOiKCNGjHjnnXfy8vK6d+9+5tHbav/617/uuOOO3Nzc+Pj4ESNGzJ8/Xwjx+uuv33PPPRde\neKHf7+/Wrdt7773XKKlOCKEE6RC++eabn3rqqYyMjMDTxMTEPXv21Bya5MSJEx07diwpKQn9\n9YqLi8ePH//EE08Ezg10OBxjx4595JFHLrjggjrbV1RUaDcVKYqSkpLidrsdDq1eXqAKkblr\nT1V9HezpBsN3nc+PTkkNZbVarVarw+HQ7veZ0Wi0WCxNNjqHIjk5WQhRXFwc60LCZ7fbXS5X\nNPskGpfJZEpISHA6ndoNRoqiJCYmNugbp6lJ8PvEay+qx4+eOcvT+QLX0OFhnBMWZVar1e/3\nn+13ZiTGuz3rrk9V9Z9s1H/28S8nCqEI36/6+/KGirOMtmGz2YIfOdS6YD12xcXFXbp0mT9/\n/oQJExRFueyyy4YMGTJhwoRAf+NPP/20bNmyvLy8Br1ecnJy//79lyxZcvfdd5tMpuXLl59/\n/vkNPVEPUaMI0T3e+lVZPR0t3Rj2DwCCUrwe9fWXxIljdc417v5OWOJceUOjXJWGKYrv8jx/\nl1z9N18qhw6KinJhtaqZ2f7eff2t28a6uFgK1mMnhPj73/8+Y8aM3NzcpUuXpqSkzJkz5913\n3z1+/LgQIiMj46abbnrkkUdqnl0YCqfTuXTp0vz8fJ/Pd8EFF0yaNKnmodha6LGLuRWVVXfv\nOxi8zfOZrUYmNmwziBp67JoCeuxijh67mDNt+cy86cPgbZxjfufLaJzjcRHShHrswtWse+yE\nELfffvu11147bdq07t27//GPf1yyZMnSpUsDuzZjuLfXtVqt06ZNC+//Ivp+l9Hy+cNHv3NV\nna1BP2vc8AaGewBobow7ttXfZue3TTzYoemrfxy7tLS0119/ffXq1S+88EKvXr22bNliNBrD\nTnXQHJNOeT2rbQdz3VfSXRhneem8NjqZf/wAwLlSXJW6kvp7rHVHD0ehGMgt1AGKr7nmmu++\n+27w4MGXXnrplClTNH1UCA3V1mTc0C7r7rSUJMPPJ/aqItWgvy899d/tzks1NPWzfQEgxqrO\netCjJqVKwxe/o4mo51Cs3+/fvn17YWGhqqpZWVnz58+/5ZZb7rjjjq5duy5ZsmTYsGHRqRIx\nZ9fr/twydU56yh6356TH09Jg6GA201EHACGJtwmdTtQ3woBq47QWnKtgPXZffvll+/bte/bs\nOWzYsGHDhnXv3r19+/ZOp/Orr7665557Ro8e/etf//rYsbov8IGUDIrSxWy61BbfyUKqA4BQ\nqQaDL4RLNX3ZjXPvATRnwYLdxIkTJ06cWFBQ4PP5/H7/vn37hg8ffuutt+r1+hkzZuzcubOs\nrKxLly5RqxUAAI1y/6p/8Aaq2ezu3is6xWiIrrHFeoUirp5x7GbPnl39tF27dvPnz1+8eLHD\n4UhISMjJyVm/fv3rr78e+SIBANA2b/tOSu++6jdfna1B1dXXqXHWaJakCRLcRC7KgkXX5OTk\n+fPnnzx5MvD06NGjDz74YHp6es2B62699dbIFggAgByG3aS/ZNCZk1WzxXX9jZ7Odd+BqZnz\nNbZYr1DEBeuxe+aZZ0aPHj1r1iyDwaCqqs/na9Omzauvvhq14gAAkIdOZ7huRFn7ToYd2/RH\nDytVLtVm956X476wp6Cv7iwafUjtZj1A8cUXX3zgwIH8/PyCggJVVTMzM3v16qVv8neyAwCg\nyfKntawadHWsq4C06hnuRK/X9+nTp0+fPtVTdu7cuW3bNrPZ3K9fv8zMzAiXBwAAgFDVc3nI\n/PnzhwwZEnhcWVk5evTobt26jRkz5uabbz7//PP/7//+L/IVAgAAICTBgt3ChQtnzZpVfez1\nL3/5y5tvvnnnnXeuXr165cqVw4YNmzdv3vLly6NSJwAAAOqhqKp6tnkdOnTo06fPihUrAk9T\nU1PHjBmzYMGC6gYjR478/vvvv//++8jVV1FRUVlZGbnlR5SiKCkpKW632+FwxLqW8CUlJZWU\nlATZTpo4q9VqtVodDofb7Y51LWEyGo0Wi0XT9/FLTk4WQhQX13+vzCbLbre7XC6PxxPrQsJk\nMpkSEhKcTmejn4oeNYqiJCYmlpSUxLqQ8CUkJJhMpqKiIk3vUf1+v8tV963PUlNTG/0VG33X\nJ/3FE8F67H766aehQ4cGHvv9/uLi4urDsgHXXXfd/v37I1gdAADA2flVsbvS9WlZ+a5Kl0+z\nibkRBQt2rVq12rZt2//a6XTZ2dnl5eU1G+zfvz/wQxwAACCaKv3qvKPHO+zY1XfXj9f9uL/f\nrh/P377rgcPHynz13JO3ifN6vYqirF+/Prz/HizYjR07dtGiRQsXLgwcw5o+ffrjjz9ene0+\n/PDDhQsXDh8+PLwXBgAACM8Jjzdv9555R46f9HirJxZ7fQuOnbhi955C9zmdNVFYWDh58uTs\n7Gyz2ZyRkTFs2LBNmzaFvbQPP/zw66+/Ppd6GiRYsPvTn/40bNiwP/zhDy1btrz22mt37dp1\n7NixnJycvLy83NzcK6+8skOHDnPnzo1arQAAAH5V3Lr/4PbKuk/1+8FVNWrfQbc/zMOyu3fv\n7tmz56effvrkk0/m5+evWLEiMTFx0KBBb7/9dngLfPLJJ5tKsDOZTKtWrVq3bt3gwYO/++67\nZ5999sCBA6dOnfrqq69SUlKWLFmyZcuWpKSkqNUKAADwz9MlX5YHuwxou7Py76eKwlv45MmT\n09LSvv7665EjR3bt2nXQoEGvvvrqrFmzduzYEWhw/Pjx0aNHt27d2mq1XnzxxZ9//rkQwu/3\nK4qyYsWKwYMHd+3aNSsr6+WXXxZCXHHFFevWrZs2bVrv3r19Pp+iKMuXL8/JyRk/fvzZFnWO\n6hmgWAgxZMiQwDUTqqpWVFQYDAaLxXLuLwwAABCGlUX1X2K/srhkYnqDr9I9efLkRx999NJL\nL9WKOjWPT95www0tWrTYtm2bzWb785//PHTo0H379qWmpur1+ieeeGLdunXp6ekvvPDC5MmT\nb7rppg8//DA7O3v27NmTJk0SQuj1+ueff/7tt9/u0KHD2RbVokWLhpZdUz0DFNekKIrNZiPV\nAQCAGNpxloOwtdqEcSw2MNZHbm7u2Rrk5+d/+eWXCxYsSE9Pt1qtf/3rX30+33vvvReYO2bM\nmPT0dCHElVde6XQ6Dx48eOYShg8f3qtXL7vdHnxRYWtAsAMAAIg5ZwjXvbr9fk+4p9l5vd6z\nzdq3b59Op+vcuXPgaVxcXFZWVnWAO++88wIPAr1gdQ7E2759+1AWFTaCHQAA0JK2ZlO9bTJM\nRpOuwQMRd+zYUVGU/Pz8WtN9Pt/ZhpX2+/3VA+CHMvSx2Ww+26yaiwobwQ4AAGjJVQm2etvk\nJdjDWHJSUtLVV1/96KOP1rpl1P3333/VVVcJITp06OD3+3ft2hWYXlFRcejQocAJcw3ViIuq\niWAHAAC0ZHJ6apwu6LAeOuWelmnhLfzpp5+urKzs0aPHihUrdu3a9cknn4wbN+7JJ5+87777\nhBDdu3cfMGDAzJkzi4qKysvLZ82aZbfbg4/pa7Va9+7de+bd8MJYVCgIdgAAQEvamkxPZbX9\n35O6DpDObdOqg+WsRzyD69ChwzfffHPVVVfdd999PXv2HD16tNPp3Lx589VXXx1osGLFCpPJ\n1LVr15ycnIMHD27atCkhISHIAu+8885nnnmmW7duZ85q6KJCoYRyK+LFixfbbLbbb7+91vRX\nXnnF4XBMmTLlHIsIoqKios5zDzVBUZSUlBS3212rR1dbkpKSSkpKNH3LaqvV6nA4zv3EhVgx\nGo0Wi6XR74QdTYF7DxYX1z9CQZNlt9tdLpfHc07D2ceQyWRKSEhwOp1OZ7DRv5oyRVESExPP\n7PbQkISEBJPJVFRUpOk9qt/vd7nqvig1NbXBw4vUK8iu793TpdN+Onzqlxc6tNDr/3Ze61HJ\nZx1k12azhXImnHaFFOwURWnTpk1hYWGt6dnZ2YcOHYroBur1eoNcnNLEKYpiNpsb5VzIGDKb\nzVVVVbGuInwGg8FgMLjdbr9fq3cP1Ol0er1eu5FC/HyysKY3JKPR6PP5NL0VmUwmre9RTSaT\nprcik8mk0+nOloo0wWAwqKrq8/nOnOXz+eLj4xv9FYP/pi3z+d8+XfJ5WfkJry9Fr+9vs96U\nnJRk0Af5L9IHu/oHKBZC/POf/7RarWdOf/bZZysqKhq7pF/wer3a/RhXBzvtroIQwmg0ut1u\n7f6+FEIYDAaPx6Pd7zODwaAoiqa3IpPJJDQe7ALZWrtbkdFoDAQ77b4LiqIYDAbt1i+EMBgM\nOp1O63tUVVXr7K2IyUrZ9brbU5NvT02O/ks3WSEFu5tuuqnO6YE7UkSUz+fTbkdF4DeB3+/X\n7ioEeDwe7e6GjEaj0PiGJH7OprGu4lxpehX8fr/X69XuKkiwO1IURVVV7dYvhAj0+Gp9j6rp\nrag5qP/iicCdxOqcdeTIkY0bNzZ2SQAAAAhHsGCnquoTTzyRkpJis9mys7OXLFlS60fGunXr\n8vLyIlwhAAAAQhIs2C1btmzGjBkZGRnjx4/PzMycMmXKzTffTAcsAABA0xTsHLtnn3120KBB\n77//vsFgEEK8+uqrd9555+233/7aa6/JfUUJAACAFgULdnv37n3iiScCqU4IMWbMmISEhJEj\nR3bo0OHBBx+MRnUAAKAZCwyw0ogLlL5nqp6rYsvLy2s+veGGGxYuXHj33Xfn5OSMGzcukoUB\nAIDmLi4uLtYlaEywYDdgwIDnnntu7NixNceSnjp16v79+ydMmFBZWVndmQcAAICYC5bMHnnk\nkf79+3fq1GnJkiWjRo2qnr5gwQKLxfL73/8+LS3MO+wCAACg0QW7KrZPnz6ffPJJbm7umePY\nzZs3b926dUlJZ70XGwAAAKKsnmOpAwYM+OSTT+qcNWTIkLy8PE3fjxkAAEAm9d954tixYydP\nnqw1ccuWLcXFxQaDoebpdwAAAIiheoLd2rVrO3fu/Prrr9eafvvtt3fu3Hnbtm0RKwwAAAAN\nEyzY7dmzZ9SoUTab7cILL6w168UXX9Tr9UOHDj19+nQkywMAAECoggW7xYsXu93ujRs3XnHF\nFbVmDRgw4L333jtx4sTixYsjWR4AAABCFSzYbdiw4cYbb+zcuXOdc3v06HHddde98cYbkSkM\nAAAADRMs2BUWFnbr1i1Ig169eh04cKCxSwIAAEA46rl4QqcL1sDv95tMpkatBwAAAGEKltty\ncnK2bt0apMEnn3ySk5PT2CUBAAAgHMGC3dChQ9esWfPNN9/UOXft2rUff/zxsGHDIlMYAAAA\nGiZYsLv33nsTExOvueaalStX+ny+6umVlZULFy68+eab09LS/vCHP0S+SAAAANQv2C3FWrZs\nuWbNmhEjRowePXrKlCndu3e32+3FxcX5+fnl5eUZGRnvvvtucnJy1GoFAABAEPXcK3bgwIE7\nd+5ctGjRmjVrPvnkE5/PZzAYunbtOnLkyKlTp5LqAAAAmo56gp0QomXLlnPnzp07d66qqk6n\n02q1KooShcoAAADQIPUHO5fLtX37do/Hk5ubm5iYGIWaAAAAEIZ6xrFbtGhRenr6RRddNHDg\nwLS0tLvuuquqqio6lQEAAKBBgvXYrVq1atq0adnZ2XfccYfVav3444+feeYZnU739NNPn8tL\n3n333QcPHqx+arFY3nzzzXNZIAAAAETwYLdw4cLs7OwdO3bYbLbAlAkTJjz//POPPPJIQkJC\n2C9ZXl4+ceLEfv36BZ4Gv7kFAAAAQhQsVOXn548ZM6Y61QkhJk2a5PF4duzYcS4vWVZWlpGR\nkfozLq0FAABoFMF67MrLy9u2bVtzSuBpeXl52K/n8Xiqqqo2b9782muvlZWVtW/ffuzYsW3a\ntAl7gQAAAAio56rYWsdJAwOdqKoa9us5nc4WLVp4vd7JkycLIVasWDFnzpxnn302Pj4+0GDl\nypXbtm0LPI6Pj58zZ47BUP+lu01T4M9lMBjsdnusawmfTqez2+3n8qbHll6vF0LExcWZzeZY\n1xImnU6n1+s1vRUFPguaXgWj0ajT6fx+f6wLCVNgZ24ymQKfCC1SFEXrH4TA11nN42CaYzAY\nVFU1Go1nzqp5hyrEUMQz02efffb4448HHs+bN69Lly6vvPJK9dxZs2aNGzfuiy++yMvLC0zZ\nuXPnxo0bA4+TkpL+/Oc/azfYBeh0Ou1GigCTyRTrEs5VnbshbdH6ViS0vwoSnBBsMBi0vkfV\n+lYkpFiFOrcij8cT/Upwpno+4fv379+yZUv10+LiYiHE7t27W7RoUT2x+jKIOvXq1WvRokWB\nxxkZGbXmxsXFpaWlnTp1qnrKvffe+/vf/z7wWKfTOZ1O7Q6woihKixYtPB7PuRy8jrmEhISy\nsjLt9thZLJa4uLjy8nLt7nQMBoPZbK6oqIh1IeELDIFZWloa60LCFx8fX1VV5fV6Y11ImIxG\no81mq6ysdLlcsa4lTIqi2O12h8MR60LCZ7PZjEZjSUmJpveoqqrW+b2sqionzTcF9QS7efPm\nzZs3r9bEP/zhDzWfBt9ArVZrVlZW9dNDhw7961//mjRpUiDvu1yukydP1gx8tTaLiooK7fbu\nVh+51u4qBPh8Pu3uhgKV+/1+7b4LOp1Ogq1IaPxIjaqqmt6KAkdgNb0hKYqi6frFz7sjre9R\nNf1BaA6CBbsHHnig0V8vOTl58+bNXq931KhRPp/vlVdesdlsAwYMaPQXAgAAaG6CBbsHH3yw\n0V/Pbrc//PDDL7300rRp04xGY6dOnebNmyfBCQcAAAAxF4OzaNu1a/fwww9H/3UBAADkpvmL\nvAAAABBAsAMAAJAEwQ4AAEASBDsAAABJEOwAAAAkQbADAACQhLZvGggAQJOiO3nCuPcH3eli\nIYQ/OcXTobM/JTXWRaEZIdgBANAIFLfb/P6/jd/vqDnRtOlDT26PqquGqEZjrApDs0KwAwDg\nXCkeT9zKl/XHj545y7hzm/50kfM3Y6JfFZohzrEDAOBcmT/+T52pLkB3uMC06aNo1oNmi2AH\nAMA5USrKjdv/G7yN8b9fiUpndOpBc0awAwDgnOgP7BN+f/A2is8n9u2JTj1ozgh2AACcE72j\nNKR2JcURLgQg2AEAcG5Ugz6kdgYujEXEEewAADgn/rSWIbXLaBXhQgCCHQAA58Z3XrYabwve\nRk1IFOflRKceNGcEOwAAzomqN7guvTJ4m6pBVwsd37mIODYyAADOlTe3e1W/gWeb677kCk/H\nLtGsB80Wd54AAKARuC+5wt+qjXnTR7pTJ6on+tJaui+7yptzfgwLQ7NCsAMAoHF423fytu+k\nO12kKy4SQvhTUv0tkmNdFJoXgh0AAI3Jn5TiT0qJdRVopgh2aJhCt2d9ecU+V5UQIsdsGpJg\nzzSyFQEA0CTwlYxQuVX1gaMnXj5d6lHV6okPHjt5W1LiQ63SLYoSw9oAAIAg2CFEbr866qfC\nTeW172DtUdWXikt2uapW5WSayHYAAMQUw50gJPNPFp2Z6qp96ax85PipaNYDAADORLBD/Rw+\n/3On6rl39fKi08U+X3TqAQAAdWrqh2Lj4uLMZnOsqzgnRqOxRYsWsa4ifDqdbqtfddU4r65O\nblXd4lNvSWlya6rT6YQQ8fHxVqs11rWESVEUnU6n9a1ICKHpVdDr9QaDQa3vg9BkKYoihLBY\nLCaTKda1hE+v12t9KxJCJCYmxrqQ8AU+yxaL5cxZXq836uWgDk092LlcrsrKylhXESZFUZKT\nkz0eT1lZWaxrCV+LFi2+Ly0NpeWuktJSszHS9TRUXFyc1Wp1Op1utzvWtYTJaDSazeby8vJY\nFxK+pKQkIURpaBtS02Sz2Vwul3a/ukwmk91u1/oeNTExUdNbUUJCgtFodDgc2v2FEBcXp6qq\ny+Wqc26dgQ9R1tSDnaqq2v0AVNP6KoQY1oxK011TTW9Igcq1W381ViGGqivX7iqIs3+QdaUl\nhh3bDEcKhMulWq2+zGxPtx6qNT76FQZX/VmW8l1AE9HUgx2agk6WkI6Gdw6tGQA0GlU1f/m5\n6YtPRI1zfA0H9pm2bHJdOcSb2z2GpQExwcUTqN8AqzXdUM9vgGS9/jJbk/t9DEBu5k0fmTZ9\nKM64cktxu+PeW2Pcnh+TqoAYItihfiad8ueMtOBt/q9lKmMUA4gm/ZFC05efBWlg3vie4tDw\nOXlAGAh2CMmoFglTUs96K+s7UlqMS9bwpWoAtMi0dXPwBorPa/rvV9EpBmgiCHYI1QMZacsy\nW59n/MWlFJkm47NtW81t1TJWVQFotvQH94XQZn8UKgGaDi6eQAMMT7TfkGjf6ar6wVWlCtHB\nbLrQYtFxABZA1ClutxLCGEb6cg2PNgWEgWCHhlGE6GYxd+MCWAAxpRoMQqcTfn/wZn4tD8gM\nhIFDsQAADdLpfOkZ9bbytWoThVqApoNgBwDQJE8Iw9R5u/WIQiVA00GwAwBokufCXr6M1sEa\ndOzizT4/avUATQHBDgCgTXp95chR/zsge8Y9rnw551cNvSH6RQGxxcUTAACtUuNtlbf+1vj1\nFuO33+h+HovYn5rm7nWRp1sPoaPzAs0OwQ4AoGGqweDuN9Ddb6BSXqarEr1RNQAAFTdJREFU\nrPTHx6tWbm+I5otgBwCQgWqz+2z2WFcBxBjd1AAAAJIg2AEAAEhCUdUzLiVCI3G5XMuWLcvO\nzr7++utjXUvztXXr1i1btlx//fXZ2dmxrqX5evHFFxVFGT9+fKwLab4OHDiwdu3a/v379+nT\nJ9a1NF/vvvvuoUOHJk6caDZz8x5ECj12EVRVVfXyyy9/8MEHsS6kWdu+ffvLL79cUFAQ60Ka\ntbfeeuutt96KdRXNWkFBwcsvv7x9+/ZYF9Ksbdy48eWXX3aHcItbIGwEOwAAAEkQ7AAAACRB\nsAMAAJAEF08AAABIgh47AAAASRDsAAAAJEGwAwAAkAT3im18xcXFL7744rfffut2u9u1azd+\n/PiOHTsKIcrLy5cuXbp9+3aPx9OpU6dJkyalp6fHuliZHT58eMGCBXv37n3nnXeqJ/IuRBl/\n8BjiIxBbfBcgJrh4ovHde++9JpNp4sSJcXFxb7zxRn5+/vLlyy0Wy1//+tfy8vI777zTbDa/\n8cYbBw8efOqpp3Q6Ok0jYtOmTcuXL+/Zs+fHH39c81uNdyHK+IPHCh+BmOO7ADHBltTIysrK\n0tLS7rrrrnbt2rVq1Wrs2LEOh6OgoODUqVNbt26dOHFiTk5O69atJ02adPjw4R07dsS6Xml5\nPJ7HH3+8X79+NSfyLkQZf/AY4iMQW3wXIFYIdo3MbrfPmTMnMzMz8LSoqEin06Wmpu7Zs8do\nNObk5ASm22y2tm3b/vDDD7GrVHJXXHFFWlparYm8C1HGHzyG+AjEFt8FiBWCXQSVlZU9/fTT\nw4cPT0pKcjgcdrtdUZTquYmJiaWlpTEsrxniXYgy/uBNDe9ITPBdgGgi2J2rzz77bPjPvv/+\n++rphYWFM2bMyM3NHTduXGBKzU8yGtfZ3oUz8S5EGX/wpoZ3JMr4LkCUcVXsuerVq9eiRYsC\njzMyMgIPvv3227/97W+jR4++7rrrAlNatGjhcDhUVa3+SJeWliYlJUW/YCnV+S6ciXchyviD\nNzW8I1HGdwGij2B3rqxWa1ZWVs0pu3bteuyxx6ZPn967d+/qiR06dPB4PPv27Wvfvr0QInAW\nbZcuXaJdrqTOfBfqxLsQZfzBmxrekWjiuwAxoX/wwQdjXYNU3G73/ffff8011/Tq1cv5M51O\nZ7fbDx069NFHH3Xq1MnpdD7zzDPx8fG33norffIRcvr06YqKikOHDm3duvWqq67iXYiJuLg4\n/uCxwkcgtvguQKwwjl0j+/bbb//85z/XmnjnnXdee+21Tqdz6dKl+fn5Pp/vggsumDRpEt3v\nkfO73/3uxIkTtaYMGzaMdyHK+IPHCh+B2OK7ALFCsAMAAJAEV8UCAABIgmAHAAAgCYIdAACA\nJAh2AAAAkiDYAQAASIJgBwAAIAmCHQAAgCQIdgAAAJIg2AFNy4MPPqj8UkJCwmWXXbZq1apa\nLY8fPz579uxu3brZ7Xa73d6lS5dp06bt2bOnVrPdu3f37t1bUZTPPvssxBpKSkqUs9i4cWOg\nTX5+/jXXXJOQkJCYmNinT5+XXnqperTzL774onfv3m3btu31/9q7+9iq6vuB46dwi/ViW1ZK\nRZ4KhjgycA/E1bIIOCphggIONBbizCCCkCUw2QPMDd3IcHNbEGiAUWdcsyAOp1UgzpDFKIoB\nRnBmi7oRo2LK4yr1YWyU9v7+OL/ddEARx0T85PX6657z/X7Publ/kDf3nHs6fPiOHTtOOPg1\n11xz0003fbgP5b9VXV09ZMiQc3MugPNB5uN+A8ApLFq06NJLL02SpL29fe/evQ0NDVOmTLnv\nvvvmzZuXTnj++ecnTpzY0tIyYcKE2traJEleeumlVatW/epXv1q/fv2ECRPSaWvWrFmwYEFZ\nWdmHOns2m62vrz9h55NPPvnEE09UVlYmSbJ9+/aRI0cOGDDg7rvv7t69+8MPPzxjxoy33377\njjvuaGtru/nmm+fOnbtw4cKlS5fW1tb+7W9/69Ll//8P+eCDD+7atevll18+i8/mQ7j55puP\nHj16bs4FcF7IAeeTu+66K0mSF154oePOd955p7Kysri4+OjRo7lcbv/+/eXl5T179ty+fXvH\naS+//HLfvn1LS0v379+fy+W2bdtWVFS0atWqtNK2bt36X7+rt99+++KLL54/f366OXr06LKy\nsgMHDqSbra2tQ4YMKSsra29v37lzZ5Ik6dDevXuTJNm9e3c67eDBgz179qyvr/+v3wYAp+dS\nLHwCFBcXT5ky5d13333ppZeSJFm+fPnhw4dXrlxZVVXVcdqQIUMaGhoWL16cfkPWq1ev7du3\nz5kz55THfOSRRwoKCurq6s7kDSxcuLCgoOCHP/xhunnLLbfU1dVVVFSkm5lMZsSIEc3Nzf/4\nxz/27t2byWTSob59+yZJ8tZbb6XT5s+fP2zYsJkzZ57mRFdddVV5efnx48c77qyuru7Tp09b\nW1uSJOvXr6+qqspmsyUlJVdcccX69es7rh01atSmTZv69+//pS99KTnpUuxp1o4aNWrkyJG7\nd++uqakpKSmpqKiora09ePBgfsKWLVtGjx5dXFzcu3fvm266ac+ePfmhZ555ZuzYsSUlJdls\ndvjw4Q888MCZfKQAHwVhB58M2Ww2SZLW1tYkSR5//PGysrJT3qk2ZsyYO+64o1evXkmSDB48\n+LOf/expDti3b9+LLrroA0/95z//ee3atUuWLCkpKUn3zJw5M73+m7dnz57Kysru3bvn/n2n\nXV6656mnnvrd7363du3agoKC05xr2rRpf//7359++un8njfffHPHjh21tbVdu3Z9+OGHa2tr\n+/Xrt2HDhoceeqhXr161tbWbN29OZ15wwQUtLS3f/va3Fy1adOedd55w5NOv7dat2xtvvDF7\n9uxFixbt2bNn9erVGzZs+M53vpOObtmyZdy4cUVFRWvWrFm6dOmuXbtGjRq1f//+JEn+8Ic/\n1NTUHDt2bN26dY8//viVV145c+bMX/ziFx/4qQJ8JD7urwyB/3DKS7G5XO6qq67KZDJHjhxp\nb2/v2rXrl7/85TM/5lleir3xxhsrKyuPHTvW2YR169YlSbJy5cpcLpf+WmLfvn25XO61115L\nkuTFF198//33Bw4cuGTJkubm5okTJ5aWlg4YMKCuru7kQx06dCiTycyaNSu/52c/+1ny7+u5\nS5cuHTNmzL/+9a90qKWlJZPJTJ8+Pd2sqalJkuTRRx/Nr73yyis//elPp6/PZO1zzz2XX1tT\nU9OnT5/09RVXXDFo0KDW1tZ0c/v27d26dVu+fHkul/vCF74wePDg999/P79w4sSJ+YvmAOeY\nsIPzSxp2mzdv3rdv3759+5qamnbs2DFjxowkSebMmZPL5d57770kSSZNmnTmxzybsPvLX/5S\nUFCwZs2aziZs3Lgxm81OmTKlvb09l8u1trb27dt38eLF7e3t3/rWtwYNGtTe3r5gwYKhQ4ce\nO3Zs9uzZw4cPP3DgwObNm7t06ZK//a6ja6+9tqKioq2tLd384he/OHTo0M7O3q9fv5EjR6av\na2pqunXr1jFAO4bdB67NZrMdR2+99dYuXbrkcrnDhw8nSTJ37tyTj3DgwIEkSebNm3e0gzVr\n1iRJsmPHjs7OC/DRcSkWzkcTJky45JJLLrnkkj59+lRVVTU0NMydO3fZsmVJkmSz2Uwm09zc\nfG7eyerVq0tKSm699dZTjq5cuXLy5MmTJk166KGH0musmUymoaGhvr6+qKho/fr1v/71r3fv\n3r1ixYr6+vrCwsLGxsaZM2dWVFSMHz9+6NChjY2NJx9z2rRpBw8efPbZZ5Mkef3113fu3HnL\nLbekQ++8887ixYsvv/zy0tLSTCaTyWTeeuut9vb2/Nry8vLCwsJTvtUPXJtev87LZDLp6L59\n+5Ikyd9Q2FFTU1OSJMuXL7+wg9tvvz3pcGchwLnkcSdwPlq2bFl6139BQUH37t2HDRvWo0eP\ndKigoOAzn/nM7t27jx49euGFF36kbyOXyz366KOTJ08uKio6efSb3/zmfffdd+eddy5ZsqTj\nnXNjxoxpamo6cuRIjx492traqqqqZs2aNWLEiNbW1gMHDgwYMCCdNmDAgFPWz+TJk7PZ7COP\nPHL11Vf/9re/LSgomDZtWjp0/fXXP//889/97ne/8pWv9OjRo6CgYNy4cR3XdlZ1Z7K2M+kv\nUTom4AlmzJhx2223nbBz8ODBZ3JwgP8tYQfno+rq6urq6s5Gv/rVr959992//OUv58+ff8LQ\nCy+88PWvf/3BBx88zfIzt23btqampmuuuebkoUWLFq1YseL+++/v7FeuaYkuW7bs4MGD99xz\nz5mf9KKLLrr++usfe+yxurq6DRs2jB49un///kmS7Nmz59lnn73tttt+/OMfpzOPHz/e3Nw8\naNCgDzzm2axNz54+uiXvjTfeyGazaaS2tbX9Tz5tgLPnUix88nzjG9/o3bv39773vfyPOlN/\n+tOfpk6d2tzcfNlll/1PTvTMM88kSTJ8+PAT9v/+97//yU9+smLFitM/u+T111+/66676urq\niouLkyQpLCzs3bv3m2++mY6+9tprAwcOPOXCadOmNTU1NTY2/vGPf8xfh01/EdyvX7/8tNWr\nV//zn/9MH4Nyemeztri4+PLLL9+0adO7776b7nnllVcGDhy4atWqsrKyqqqqxsbGI0eO5Oc3\nNDR8//vfP+GJLQDnhm/s4JOnZ8+eTzzxxIQJE6677rqampqRI0d27dr1xRdfbGxsLC8vf+qp\np9I/NfHcc8+98sor6YskSTZt2pRujhs3rn///k8++eScOXN+9KMffe1rX+vsROn89G9g5B0/\nfnzevHnl5eUXXHDB/fff33Fo/Pjxffr0yW/efvvt48ePnzRpUn7PlClT6uvrp06dum3btldf\nffWGG2445XmvvfbasrKyBQsWFBUVTZ06Nd05ePDg/v37r1279vOf/3zPnj0fe+yxXbt2XX31\n1bt27Xr66adPeKTfCc5mbZIk99xzz8SJE8eOHTtv3rz33nvv5z//eUVFxezZs5Mkuffee8eO\nHTt69OgFCxb07t1769atP/3pT6dPn57J+NcV+Dh83L/eAP5DZ487Odnhw4cXLlw4bNiw7t27\nFxcXf+5zn/vBD35w6NCh/ITOvk7buHFjLpfbsGFD8u9nlHRm/PjxhYWFJ+w8dOhQZ/+ebNmy\nJT/tN7/5TWlpaVNTU8e1LS0t06dPLy0tvfTSS9etW3eaU8+aNStJkhtvvLHjzp07d44YMSKb\nzV588cWzZ89uaWnZuHFjeXn5pz71qVdffbWmpqaysrLj/I6/iv2wa9NPL7+5efPm6urqbDZb\nUVFxww03/PWvf80Pbd26dezYscXFxYWFhZdddtm9996bfzAKwDlWkDvpaaIAAHwSuccOACAI\nYQcAEISwAwAIQtgBAAQh7AAAghB2AABBCDsAgCCEHQBAEMIOACAIYQcAEISwAwAIQtgBAAQh\n7AAAghB2AABBCDsAgCCEHQBAEMIOACAIYQcAEISwAwAIQtgBAAQh7AAAghB2AABBCDsAgCCE\nHQBAEMIOACAIYQcAEISwAwAIQtgBAAQh7AAAghB2AABBCDsAgCCEHQBAEMIOACAIYQcAEISw\nAwAIQtgBAAQh7AAAghB2AABBCDsAgCCEHQBAEMIOACAIYQcAEISwAwAIQtgBAAQh7AAAghB2\nAABBCDsAgCCEHQBAEMIOACAIYQcAEISwAwAIQtgBAAQh7AAAghB2AABBCDsAgCCEHQBAEMIO\nACAIYQcAEISwAwAIQtgBAAQh7AAAghB2AABBCDsAgCCEHQBAEMIOACAIYQcAEISwAwAIQtgB\nAAQh7AAAghB2AABBCDsAgCCEHQBAEMIOACAIYQcAEISwAwAIQtgBAAQh7AAAghB2AABBCDsA\ngCCEHQBAEMIOACAIYQcAEISwAwAIQtgBAAQh7AAAghB2AABBCDsAgCCEHQBAEMIOACAIYQcA\nEISwAwAIQtgBAAQh7AAAghB2AABBCDsAgCCEHQBAEMIOACAIYQcAEISwAwAIQtgBAAQh7AAA\nghB2AABBCDsAgCCEHQBAEMIOACAIYQcAEISwAwAIQtgBAAQh7AAAghB2AABBCDsAgCCEHQBA\nEMIOACAIYQcAEISwAwAIQtgBAAQh7AAAghB2AABBCDsAgCCEHQBAEMIOACAIYQcAEISwAwAI\nQtgBAAQh7AAAghB2AABBCDsAgCCEHQBAEMIOACAIYQcAEISwAwAIQtgBAAQh7AAAghB2AABB\nCDsAgCCEHQBAEMIOACAIYQcAEISwAwAIQtgBAAQh7AAAghB2AABBCDsAgCCEHQBAEMIOACAI\nYQcAEISwAwAIQtgBAAQh7AAAghB2AABBCDsAgCCEHQBAEMIOACAIYQcAEISwAwAIQtgBAAQh\n7AAAghB2AABBCDsAgCCEHQBAEMIOACAIYQcAEISwAwAIQtgBAAQh7AAAghB2AABBCDsAgCCE\nHQBAEMIOACAIYQcAEISwAwAIQtgBAAQh7AAAghB2AABBCDsAgCCEHQBAEMIOACAIYQcAEISw\nAwAIQtgBAAQh7AAAghB2AABBCDsAgCCEHQBAEMIOACAIYQcAEISwAwAIQtgBAAQh7AAAghB2\nAABBCDsAgCCEHQBAEMIOACAIYQcAEISwAwAIQtgBAAQh7AAAghB2AABBCDsAgCCEHQBAEMIO\nACAIYQcAEISwAwAIQtgBAAQh7AAAghB2AABBCDsAgCCEHQBAEMIOACAIYQcAEISwAwAIQtgB\nAAQh7AAAghB2AABBCDsAgCCEHQBAEMIOACAIYQcAEISwAwAIQtgBAAQh7AAAghB2AABBCDsA\ngCCEHQBAEMIOACAIYQcAEISwAwAIQtgBAAQh7AAAghB2AABBCDsAgCCEHQBAEMIOACAIYQcA\nEISwAwAIQtgBAAQh7AAAghB2AABBCDsAgCCEHQBAEMIOACAIYQcAEISwAwAIQtgBAAQh7AAA\nghB2AABBCDsAgCCEHQBAEMIOACAIYQcAEISwAwAIQtgBAAQh7AAAghB2AABBCDsAgCCEHQBA\nEMIOACAIYQcAEISwAwAIQtgBAAQh7AAAghB2AABBCDsAgCCEHQBAEMIOACAIYQcAEISwAwAI\nQtgBAAQh7AAAghB2AABBCDsAgCCEHQBAEMIOACAIYQcAEISwAwAIQtgBAAQh7AAAghB2AABB\nCDsAgCCEHQBAEMIOACAIYQcAEISwAwAI4v8AZtwPvexYraoAAAAASUVORK5CYII=",
      "text/plain": [
       "plot without title"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "plotPCA(dat, intgroup=\"Sample_Group\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R [conda env:analysis]",
   "language": "R",
   "name": "conda-env-analysis-r"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.5.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
